plugins/inputs/t128_graphql/t128_graphql.go

```diff
@@ -174,26 +174,47 @@ func (plugin *T128GraphQL) Gather(acc telegraf.Accumulator) error {
 }
 
 func (plugin *T128GraphQL) processResponse(jsonChildren []*gabs.Container, acc telegraf.Accumulator) {
-	//TODO: allow nested fields/tags - MON-310
 	for _, child := range jsonChildren {
 		node := child.Data().(map[string]interface{})
 		fields := make(map[string]interface{})
 		tags := make(map[string]string)
 
 		for fieldRenamed, fieldName := range plugin.Fields {
-			if isNil(node[fieldName]) {
+			data := node[fieldName]
+
+			if strings.Contains(fieldName, "/") {
+				nestedObj, err := child.JSONPointer("/" + fieldName)
```

Comment:
This `child.JSONPointer` seems to be how to execute a query, but it looks like you are using it to go one more deep. Could this be refactored so that the parsed response is just queried by this method for the whole field path? In sudo code:
```
response = getResponse()
jsonResponse = gabs.Parse(response)
baseJsonResponse = gabs.JSONPointer(baseUrlPath)
for field in fields:
     # maybe need to convert the field path into gabs syntax
     fieldElem = baseJsonResponse.JSONPointer(field)
     fields[fieldRenamed] = fieldElem.Data()

for tag in tags:
    # pretty much the same logic as fields
```
Obvs this is an oversimplication, but it seems that the current code is limiting itself to a depth of 2